test/calltoaction.spec.js

```diff
@@ -1,151 +1,114 @@
-        callToAction.handleInteraction(getMockEvent('call-to-action__button'));
-        expect(utils.getCookieValue(expectedCookieName, document.cookie)).to.equal('true');
-      });
+  describe('dismiss button', function () {
 
+    it('is a child of .call-to-action', function () {
+      const callToAction = new CallToAction(fixtures.generateHTML());
+      const $button = callToAction.dismissible.$button;
+      expect($button.parentElement.classList.contains('call-to-action')).to.be.true;
     });
 
-    context(`when the cookie "${expectedCookieName}=true" has previously been set`, () => {
+    context('when clicked', function () {
 
-      let callToAction;
-
-      beforeEach(() => {
-        const expiryDate = 'Tue, 19 January 2038 03:14:07 UTC';
-        document.cookie = `${expectedCookieName}=true; expires=${expiryDate}; path=/;`;
-        document.querySelector('.call-to-action-wrapper').classList.add('call-to-action-wrapper--js-shown');
+      it('adds CSS class "hidden" to the call to action component', function () {
+        const $fixture = fixtures.generateHTML();
+        const callToAction = new CallToAction($fixture);
+        const $button = callToAction.dismissible.$button;
+        expect($fixture.classList.contains('hidden')).to.be.false;
+        $button.click();
+        expect($fixture.classList.contains('hidden')).to.be.true;
       });
 
-      it('does not have the CSS class "call-to-action-wrapper--js-shown"', () => {
-        callToAction = new CallToAction($elm);
-        expect($elm.classList.contains('call-to-action-wrapper--js-shown')).to.be.false;
-      });
-
-    });
-
-    context(`when the cookie "${expectedCookieName}=true" is not currently set`, () => {
-
-      let callToAction;
+      describe('sets a cookie with', function () {
 
-      beforeEach(() => {
-        expireCookie(expectedCookieName);
-        document.querySelector('.call-to-action-wrapper').classList.remove('call-to-action-wrapper--js-shown');
-      });
+        describe('name', function () {
 
-      it('has the CSS class "call-to-action-wrapper--js-shown"', () => {
-        callToAction = new CallToAction($elm);
-        expect($elm.classList.contains('call-to-action-wrapper--js-shown')).to.be.true;
-      });
+          it('is derived from the HTML element\'s id and data-cookie-name-root attribute', function () {
+            const id = fixtures.generateRandomId();
+            const $fixture = fixtures.generateHTMLWithCookieDetails(fixtures.getCookieNameRoot(), id);
+            const callToAction = new CallToAction($fixture);
+            callToAction.dismissible.$button.click();
+            const expectedCookieName = `${fixtures.getCookieNameRoot()}${id}`;
+            expect(callToAction.dismissible.cookieName).to.equal(expectedCookieName);
+            expect(utils.getCookieValue(expectedCookieName, document.cookie)).to.equal('true');
+            fixtures.clearCookie(id);
```

Comment:
not sure if a failed `expect` interrupts the test, but shoudn't clean up code be in an `afterEach()` in general? (wrapped in an `if` in case it were not always needed)

Reply:
Not modifying this without understanding the reason for its placement.

Reply:
Looking at the other tests, it looks more like a missing `expect`.